tests/test_parsers.py

```diff
@@ -410,3 +414,99 @@ def test_round_trip(self):
+            "predicate_id",
+            "object_id",
+            "object_label",
+            "mapping_justification",
+            "creator_id",
+        ]
+
+        df = pd.DataFrame(rows, columns=columns)
+        msdf = MappingSetDataFrame(df=df, converter=ensure_converter())
+        msdf.clean_prefix_map(strict=True)
+
+        #: This is a set of the prefixes that explicitly are used in this
+        #: example. SSSOM-py also adds the remaining builtin prefixes from
+        #: :data:`sssom.context.SSSOM_BUILT_IN_PREFIXES`, which is reflected
+        #: in the formulation of the test expectation below
+        explicit_prefixes = {"DOID", "semapv", "orcid", "skos", "UMLS"}
+        self.assertEqual(
+            explicit_prefixes.union(SSSOM_BUILT_IN_PREFIXES),
+            set(msdf.prefix_map),
+        )
+
+        with TemporaryDirectory() as dir:
+            directory = Path(dir)
+            path = directory.joinpath("test.sssom.x")
+            path_str = str(path)
+            with path.open("w") as file:
+                write_json(msdf, file)
+
+            reconsitited_msdf = parse_sssom_json(path_str)
+            reconsitited_msdf.clean_prefix_map(strict=True)
+
+            test_meta = {
+                "mapping_set_title": "A title",
+                "license": "https://w3id.org/sssom/license/test",
+            }
+
+            reconsitited_msdf_with_meta = parse_sssom_json(path_str, meta=test_meta)
+            reconsitited_msdf_with_meta.clean_prefix_map(strict=True)
+
+        # Ensure the prefix maps are equal after json parsing and cleaning
+        self.assertEqual(
+            msdf.prefix_map,
+            reconsitited_msdf.prefix_map,
+        )
+
+        # Ensure the shape, labels, and values in the data frame are the same after json parsing and cleaning
+        self.assertTrue(
+            msdf.df.equals(reconsitited_msdf.df),
+        )
+
+        # Ensure the metadata is the same after json parsing and cleaning
+        self.assertEqual(
+            msdf.metadata,
+            reconsitited_msdf.metadata,
+        )
+
+        combine_meta = dict(
+            ChainMap(
+                msdf.metadata,
+                test_meta,
+            )
+        )
+
+        # Ensure the metadata after json parsing with additional metadata corresponds to
+        # a chain of the original metadata with the test metadata.
+        # In particular, this ensures that fields in the test metadata provided are added
+        # to the MappingSet if they are not present, but not updated if they are already present.
+        self.assertEqual(
+            combine_meta,
+            reconsitited_msdf_with_meta.metadata,
+        )
+
+    def test_round_trip_json_tsv(self):
+        """Test JSON => TSV using parse()."""
+        self._basic_round_trip(parse_sssom_json)
+
+    def test_round_trip_rdf_tsv(self):
+        """Test RDF => TSV using parse()."""
+        self._basic_round_trip(parse_sssom_rdf)
```

Comment:
@cthoyt I added the test - as it is identical with the json one, I decided to not duplicate the code. Do you consider this an anti-pattern?

Reply:
This is great. It means the test can be arbitrarily extended for all formats using the _registry_ pattern. I updated the test in [2081f47](https://github.com/mapping-commons/sssom-py/pull/480/commits/2081f47cfec98e6ca3a3f3acf13382b586a1e812) to do exactly that (though now it identifies more errors to fix ;)